shared/src/baseline_solver.rs

```diff
@@ -10,6 +10,11 @@ const DEFAULT_MAX_HOPS: usize = 2;
 
 type PathCandidate = Vec<H160>;
 
+/// Note that get_amount_out and get_amount_in are not always symmetrical. That is for some AMMs it
+/// is possible that get_amount_out returns an amount for which get_amount_in returns None when
+/// trying to go the reverse direction. Or that the resulting amount is different from the original.
+/// This situation is rare and resulting amounts should usually be identical or very close but it
+/// can occur.
```

Comment:
Could we, perhaps, introduce a unit test which demonstrates that these scenarios described are indeed possible? It would help me and the reader to have a sense of when this can happen? My only guess at the moment (if this is rare) would be when pool reserves are very low on one side and very high on the other... Or possibly when the amount in is greater than the reserves the pool currently holds of that token.

Reply:
My assumption is that this is possible if the traded amounts are very small (e.g. 1wei). In this case out amount may return 0 or one, while get_amount_in for that value returns None.

Agree that a unit test demonstrating the issue for one specific Pool would be a nice way of improving our understanding of it.

Reply:
On this note (I forgot if I mentioned this somewhere last week), maybe it would make sense to reintroduce fuzzing to components like the BaselineSolver. That would likely find example cases quite easily